src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,8 +1,22 @@
-export const TodoFilter = () => (
+import React from 'react';
+
+type Props = {
+  onFilterKeyChange: (key: string) => void,
+  onQueryChange: (event: React.ChangeEvent<HTMLInputElement>) => void,
+  onDelete: () => void
+  query: string,
```

Comment:
It is really cool that you added some optimization features. Although, you pass some props which are not wrapped in `useMemo()`. The question is whether this optimization features needed.